pkgs/tools/misc/bat-extras/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv, fetchFromGitHub, bash, less, makeWrapper, ncurses, bat, ripgrep
+, withShFmt ? true, shfmt
+, withPrettier ? true, nodePackages
+, withClangTools ? true, clang-tools
+, withRustFmt ? true, rustfmt
+, withEntr ? true, entr
+}:
+
+stdenv.mkDerivation rec {
+  pname   = "bat-extras";
+  version = "20200408";
+
+  src = fetchFromGitHub {
+    owner  = "eth-p";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "184d5rwasfpgbj2k98alg3wy8jmzna2dgfik98w2a297ky67s51v";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ bash makeWrapper ] ++ stdenv.lib.optional withShFmt shfmt;
+
+  phases = [ "unpackPhase" "buildPhase" "installPhase" ];
+
+  buildPhase = ''
+    patchShebangs test.sh
+    patchShebangs .test-framework/bin/best.sh
+
+    chmod u+x .test-framework/lib/print_util.sh
+    wrapProgram .test-framework/lib/print_util.sh \
+      --prefix PATH : "${ncurses}/bin" \
+
+    bash ./build.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp bin/* $out/bin
+
+    substituteInPlace $out/bin/batgrep \
+      --replace '$PAGER' '${less}/bin/less'
```

Comment:
This is going to force pager behavior always, when the pager would otherwise not be used. Not only that, but it also disables the normal reading of the `PAGER` env var.

By that I mean, if I build bat-extras myself and run `batgrep` with my current environment, it doesn't use a pager, because I don't have any pager-related env vars set up. But with this substitution, it will completely ignore the `PAGER` env var and always use Nix-installed `less`, even if I wanted to use something else.

What we do for `bat` is just inject Nix's `less` into `PATH`, so when `bat` defaults to `less` (or when `PAGER=less`) it uses the Nix `less` (I forget why this was important but I think it works around some bug in macOS `less` combined with Bat's default options). I'm inclined to say we should just do the same here and wrap each of the bat-extras scripts such that Nix's `less` is prefixed to `PATH`. This way `PAGER=less batgrep foo` will use the same pager that `PAGER=less bat foo` uses.